File: VoCat.UI/vocat_mobile/lib/models/folder.dart
class Folder {
  final String folderId;
  final String name;
  final String? parentFolderId;
  final String userId;
  final DateTime createdAt;
  final DateTime? updatedAt;

  Folder({
    required this.folderId,
    required this.name,
    this.parentFolderId,
    required this.userId,
    required this.createdAt,
    this.updatedAt,
  });

  factory Folder.fromJson(Map<String, dynamic> json) {
    return Folder(
      folderId: json['folderId'],
      name: json['name'],
      parentFolderId: json['parentFolderId'],
      userId: json['userId'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt:
          json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() => {
    "folderContract": {
      "folderId": folderId,
      "name": name,
      "parentFolderId": parentFolderId?.toString(),
      "userId": userId,
      "createdAt": createdAt.toIso8601String(),
      "updatedAt": updatedAt?.toIso8601String(),
    },
  };
}

File: VoCat.Types/Folder.cs
using VoCat.SharedKernel;

namespace VoCat.Types;

public class Folder
{
    public Folder()
    {
        folderId = Guid.NewGuid();
        createdAt = DateTime.UtcNow;
    }

    #region FolderId
    private Guid folderId;
    public Guid FolderId
    {
        get => folderId;
        set
        {
            GuardClause.EnsureNotNullOrEmpty(value, nameof(FolderId));
            folderId = value;
        }
    }
    #endregion
    #region Name
    private string name;
    public string Name{
        get => name;
        set
        {
           GuardClause.EnsureNotNullOrEmpty(value,nameof(Name)); 
           name = value;
        }
    }
    #endregion
    #region ParenFolderId
    private Guid? parentFolderId; 
    public Guid? ParentFolderId
    {
        get => parentFolderId;
        set
        {
            parentFolderId = value;
        }
    }
    #endregion
    #region UserId
    private Guid userId; 
    public Guid UserId
    {
        get => userId;
        set
        {
            GuardClause.EnsureNotNullOrEmpty(value, nameof(UserId));
            userId = value;
        }
    }
    #endregion
    #region CreatedAt
    private DateTime createdAt;
    public DateTime CreatedAt{
        get => createdAt;
        set
        {
            GuardClause.EnsureNotNullOrEmpty(value,nameof(CreatedAt));
            createdAt = value;
        }
    }
    #endregion
    #region UpdatedAt
    private DateTime? updatedAt;
    public DateTime? UpdatedAt{
        get => updatedAt;
        set
        {
            updatedAt = value;
        }
    }
    #endregion
}
